spec/downloader/base_spec.rb

```diff
@@ -135,6 +135,40 @@
     end
   end
 
+  context 'with download_retry_count' do
+    before do
+      stub_request(:get, uri).to_return(status: 503)
+    end
+    let(:instance) { described_class.new(uploader) }
+    subject { instance.download uri, download_retry_count }
+    context 'when failed' do
+      context 'when download_retry_count == 0 ' do
+        let(:download_retry_count) { 0 }
+        it { expect { subject }.to raise_error CarrierWave::DownloadError }
+
+        it do
+          begin
+            subject
+          rescue CarrierWave::DownloadError
+            expect(instance.current_download_retry_count).to eq 0
+          end
+        end
+      end
+
+      context 'when download_retry_count > 0' do
+        let(:download_retry_count) { 1 }
+        it { expect { subject }.to raise_error CarrierWave::DownloadError }
+        it do
+          begin
+            subject
+          rescue CarrierWave::DownloadError
+            expect(instance.current_download_retry_count).to eq 1
+          end
+        end
```

Comment:
Testing a certain behavior using its internal state (`current_download_retry_count`) is not a good idea. Instead, you can use this API to imitate download attempts which fails initially but successes in the next time.
https://github.com/bblimke/webmock#multiple-responses-for-repeated-requests